integrationlink/params.go

```diff
@@ -0,0 +1,133 @@
+package integrationlink
+
+import (
+	"gopkg.in/errgo.v1"
+
+	"github.com/urfave/cli"
+
+	"github.com/Scalingo/cli/config"
+	"github.com/Scalingo/go-scalingo"
+)
+
+func CheckAndFillParams(c *cli.Context, app string) (*scalingo.SCMRepoLinkParams, error) {
+	sc, err := config.ScalingoClient()
+	if err != nil {
+		return nil, errgo.Notef(err, "fail to get Scalingo client")
+	}
+
+	repoLink, err := sc.SCMRepoLinkShow(app)
+	if err != nil {
+		return nil, errgo.Notef(err, "fail to get repo link")
+	}
+
+	paramsChecker := newParamsChecker(repoLink, c)
+	params := &scalingo.SCMRepoLinkParams{
+		Branch:                   paramsChecker.lookupBranch(),
+		AutoDeployEnabled:        paramsChecker.lookupAutoDeploy(),
+		DeployReviewAppsEnabled:  paramsChecker.lookupDeployReviewApps(),
+		DestroyOnCloseEnabled:    paramsChecker.lookupDestroyOnClose(),
+		HoursBeforeDeleteOnClose: paramsChecker.lookupHoursBeforeDestroyOnClose(),
+		DestroyStaleEnabled:      paramsChecker.lookupDestroyOnStale(),
+		HoursBeforeDeleteStale:   paramsChecker.lookupHoursBeforeDestroyOnStale(),
+	}
+
+	return params, nil
+}
+
+type paramsChecker struct {
+	repoLink *scalingo.SCMRepoLink
+	ctx      *cli.Context
+}
+
+func newParamsChecker(repoLink *scalingo.SCMRepoLink, ctx *cli.Context) *paramsChecker {
+	return &paramsChecker{repoLink: repoLink, ctx: ctx}
+}
+
+func (p *paramsChecker) lookupBranch() *string {
+	branch := p.ctx.String("branch")
+
+	if branch != "" && p.repoLink.Branch != branch {
```

Comment:
Why are those test needed? if we don't do them, we do not need to show the RepoLink which will remove the Show API call.